// @tiptap/extension-hard-break/dist/tiptap-extension-hard-break.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('@tiptap/core');
var prosemirrorCommands = require('prosemirror-commands');

const HardBreak = core.Node.create({
    name: 'hardBreak',
    defaultOptions: {
        HTMLAttributes: {},
    },
    inline: true,
    group: 'inline',
    selectable: false,
    parseHTML() {
        return [
            { tag: 'br' },
        ];
    },
    renderHTML({ HTMLAttributes }) {
        return ['br', core.mergeAttributes(this.options.HTMLAttributes, HTMLAttributes)];
    },
    addCommands() {
        return {
            setHardBreak: () => ({ commands, state, dispatch }) => {
                return commands.first([
                    () => prosemirrorCommands.exitCode(state, dispatch),
                    () => {
                        if (dispatch) {
                            state.tr.replaceSelectionWith(this.type.create()).scrollIntoView();
                        }
                        return true;
                    },
                ]);
            },
        };
    },
    addKeyboardShortcuts() {
        return {
            'Mod-Enter': () => this.editor.commands.setHardBreak(),
            'Shift-Enter': () => this.editor.commands.setHardBreak(),
        };
    },
});

exports.HardBreak = HardBreak;
exports.default = HardBreak;
