// @tiptap/extension-hard-break/dist/tiptap-extension-hard-break.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@tiptap/core'), require('prosemirror-commands')) :
  typeof define === 'function' && define.amd ? define(['exports', '@tiptap/core', 'prosemirror-commands'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['@tiptap/extension-hard-break'] = {}, global.core, global.prosemirrorCommands));
}(this, (function (exports, core, prosemirrorCommands) { 'use strict';

  const HardBreak = core.Node.create({
      name: 'hardBreak',
      defaultOptions: {
          HTMLAttributes: {},
      },
      inline: true,
      group: 'inline',
      selectable: false,
      parseHTML() {
          return [
              { tag: 'br' },
          ];
      },
      renderHTML({ HTMLAttributes }) {
          return ['br', core.mergeAttributes(this.options.HTMLAttributes, HTMLAttributes)];
      },
      addCommands() {
          return {
              setHardBreak: () => ({ commands, state, dispatch }) => {
                  return commands.first([
                      () => prosemirrorCommands.exitCode(state, dispatch),
                      () => {
                          if (dispatch) {
                              state.tr.replaceSelectionWith(this.type.create()).scrollIntoView();
                          }
                          return true;
                      },
                  ]);
              },
          };
      },
      addKeyboardShortcuts() {
          return {
              'Mod-Enter': () => this.editor.commands.setHardBreak(),
              'Shift-Enter': () => this.editor.commands.setHardBreak(),
          };
      },
  });

  exports.HardBreak = HardBreak;
  exports.default = HardBreak;

  Object.defineProperty(exports, '__esModule', { value: true });

})));
